apis/v1alpha1/gateway_types.go

```diff
@@ -125,20 +125,26 @@ type GatewaySpec struct {
 // combination of Hostname, Port, and Protocol. This will be enforced by a
 // validating webhook.
 type Listener struct {
-	// Hostname specifies to match the virtual hostname for
-	// protocol types that define this concept.
+	// Hostname specifies the virtual hostname to match for protocol types that
+	// define this concept. When unspecified or "*", all hostnames are matched.
 	//
-	// Incoming requests that include a hostname are matched
-	// according to the given HostnameMatchType to select
-	// the Routes from this Listener.
+	// Hostname is the fully qualified domain name of a network host, as defined
+	// by RFC 3986. Note the following deviations from the "host" part of the
+	// URI as defined in the RFC:
 	//
-	// If a match type other than "Any" is supplied, it MUST
-	// be compatible with the specified Protocol field.
+	// 1. IP literals are not allowed.
+	// 2. The `:` delimiter is not respected because ports are not allowed.
+	//
+	// Hostname can be "precise" which is a domain name without the terminating
+	// dot of a network host (e.g. "foo.example.com") or "wildcard", which is a
+	// domain name prefixed with a single wildcard label (e.g. "*.example.com").
+	// The wildcard character '*' must appear by itself as the first DNS label
+	// and matches only a single label.
 	//
 	// Support: Core
 	//
-	// +kubebuilder:default={match: "Any"}
-	Hostname HostnameMatch `json:"hostname,omitempty"`
+	// +optional
```

Comment:
I remember Tim had a comment in pre-alpha review that we should explicitly add in a line for protocols that don't care about or don't have the concept of hostname.

We currently do have:
```
When unspecified or "*", all hostnames are matched
```

Should we add in something along the lines of "This field can be omitted for protocols that don't require hostname based matching".

Reply:
Updated, thanks!